main/lsp/requests/references.cc

```diff
@@ -43,11 +43,14 @@ LSPResult LSPLoop::handleTextDocumentReferences(unique_ptr<core::GlobalState> gs
                     response->result = extractLocations(*gs, run2.responses);
                 }
             } else if (auto identResp = resp->isIdent()) {
-                std::vector<std::shared_ptr<core::File>> files;
-                auto run2 = runLSPQuery(
-                    move(gs), core::lsp::Query::createVarQuery(identResp->owner, identResp->variable), files, true);
-                gs = move(run2.gs);
-                response->result = extractLocations(*gs, run2.responses);
+                auto loc = identResp->owner.data(*gs)->loc();
+                if (loc.exists()) {
+                    auto run2 =
+                        runLSPQuery(move(gs), core::lsp::Query::createVarQuery(identResp->owner, identResp->variable),
```

Comment:
AFAIK, references to local variables cannot span multiple files, so now I restrict local variable queries to the file that defines the symbol that owns it.

If I'm wrong about this, let me know...

Reply:
AFAIK you're right :-)